README.md

```diff
@@ -265,6 +265,154 @@ end)
+					heading = GetEntityHeading(entity)
+				end
+				if entity and IsControlPressed(0, 38) then break end
+			end
+			Wait(0)
+		end
+		SetEntityDrawOutline(entity, false)
+		if not model or model == 0 then QBCore.Functions.Notify(Lang:t("error.door_not_found"), 'error') canContinue = true return end
+		result = DoorSystemFindExistingDoor(coords.x, coords.y, coords.z, model)
+		if result then QBCore.Functions.Notify(Lang:t("error.door_registered"), 'error') canContinue = true return end
+		doorData.doorHash = 'door_'..doorData.dooridentifier
+		AddDoorToSystem(doorData.doorHash, model, coords, false, false, false)
+		DoorSystemSetDoorState(doorData.doorHash, 4, false, false)
+		coords = GetEntityCoords(entity)
+		heading = GetEntityHeading(entity)
+		RemoveDoorFromSystem(doorData.doorHash)
+		doorData.entity = entity
+		doorData.coords = coords
+		doorData.model = model
+		doorData.heading = heading
+		TriggerServerEvent('qb-doorlock:server:saveNewDoor', doorData, false)
+		canContinue = true
+	else
+		local result, entityHit
+		local entity, coords, heading, model = {0, 0}, {0, 0}, {0, 0}, {0, 0}
+		for i = 1, 2 do
+			QBCore.Functions.Notify("Point any Weapon at the Door. Press [E] to confirm it.", 'primary', 30000)
+			while true do
+				if IsPlayerFreeAiming(PlayerId()) then
+					result, entityHit = raycastWeapon()
+					if result and entityHit ~= entity[i] then
+						SetEntityDrawOutline(entity[i], false)
+						SetEntityDrawOutline(entityHit, true)
+						entity[i] = entityHit
+						coords[i] = GetEntityCoords(entity[i])
+						model[i] = GetEntityModel(entity[i])
+						heading[i] = GetEntityHeading(entity[i])
+					end
+					if entity[i] and IsControlPressed(0, 38) then break end
+				end
+				Wait(0)
+			end
+			Wait(200)
+		end
+		SetEntityDrawOutline(entity[1], false)
+		SetEntityDrawOutline(entity[2], false)
+		if not model[1] or model[1] == 0 or not model[2] or model[2] == 0 then QBCore.Functions.Notify(Lang:t("error.door_not_found"), 'error') return end
+		if entity[1] == entity[2] then QBCore.Functions.Notify(Lang:t("error.same_entity"), 'error') canContinue = true return end
+		doorData.doorHash = {}
+		for i = 1, 2 do
+			result = DoorSystemFindExistingDoor(coords[i].x, coords[i].y, coords[i].z, model[i])
+			if result then QBCore.Functions.Notify(Lang:t("error.door_registered"), 'error') canContinue = true return end
+			doorData.doorHash[i] = 'door_'..doorData.dooridentifier..'_'..i
+			AddDoorToSystem(doorData.doorHash[i], model[i], coords[i], false, false, false)
+            DoorSystemSetDoorState(doorData.doorHash[i], 4, false, false)
+            coords[i] = GetEntityCoords(entity[i])
+            heading[i] = GetEntityHeading(entity[i])
+            RemoveDoorFromSystem(doorData.doorHash[i])
+		end
+		doorData.entity = entity
+		doorData.coords = coords
+		doorData.model = model
+		doorData.heading = heading
+		TriggerServerEvent('qb-doorlock:server:saveNewDoor', doorData, true)
+		canContinue = true
```

Comment:
same comment here. looks like a lot of this is duplicate from above, lets condense it down into a function and remove code duplication

Reply:
same as above is guess, difference between saveNewDoor true/false